pynumdiff/kalman_smooth/_kalman_smooth.py

```diff
@@ -6,638 +6,255 @@
 ####################
 # Helper functions #
 ####################
-
-
-def __kalman_forward_update__(xhat_fm, P_fm, y, u, A, B, C, R, Q):
```

Comment:
I found this function unnecessary, so its logic now lives in the forward filter.